### PR TITLE
feat: Resolving API groups

### PR DESCRIPTION
**Pull request by danielpacak**
Resolves: #38 , #35

### CODE REVIEW

File: pkg/cmd/policy_rule_matcher_test.go [hunk 1]
```diff
@@ -15,15 +16,20 @@ func TestMatcher_MatchesRole(t *testing.T) {
 		Rules: []rbac.PolicyRule{
 			{
 				Verbs:     []string{"get", "list"},
+				APIGroups: []string{""},
```
Comment (lizrice) [inline pkg/cmd/policy_rule_matcher_test.go]: Given you've gone to the trouble of adding (empty) APIGroups into these tests I wonder if it's worth have a case that is non-empty, more to catch regressions / document what is supposed to happen than for code coverage. Maybe not really necessary because you have tests for matching against individual rules in `TestMatcher_matches`. Wdyt?

File: test/integration_test.go [hunk 2]
```diff
@@ -101,10 +121,32 @@ func prettyPrintWhoCanOutput(t *testing.T, args []string, out *bytes.Buffer) {
...
+func createCRDs(t *testing.T, client clientext.CustomResourceDefinitionInterface) {
 	t.Helper()
+	_, err := client.Create(&apiext.CustomResourceDefinition{
+		ObjectMeta: meta.ObjectMeta{
+			Name: "pods.metrics.k8s.io",
+		},
+		Spec: apiext.CustomResourceDefinitionSpec{
+			Scope: apiext.NamespaceScoped,
+			Group: "metrics.k8s.io",
+			// TODO Use Versions filed to be a good citizen
```
Comment (lizrice) [inline test/integration_test.go]: TODO here (which I don't totally understand!)

Comment (danielpacak) [reply test/integration_test.go]: My bad. The Version property is deprecated. I've changed it to:

```go
Versions: []apiext.CustomResourceDefinitionVersion{
	{
		Name:    "v1beta1",
		Served:  true,
		Storage: true,
	},
},
```

File: pkg/cmd/policy_rule_matcher_test.go (hunk 1 above)
Comment (danielpacak) [reply pkg/cmd/policy_rule_matcher_test.go]: Definitely worth it. Thanks! I've added the case with `deployments.extensions`.

File: pkg/cmd/resource_resolver.go [hunk 3]
```diff
@@ -72,19 +102,11 @@ func (rv *resourceResolver) resourceFor(resourceArg, subResource string) (apisme
 }
```
[the remark this answers is not in the document]
Comment (danielpacak) [reply pkg/cmd/resource_resolver.go]: Done
